fuzzing/instrum_opts.bzl

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Fuzz test instrumentation options.
+
+Each fuzzing engine or sanitizer instrumentation recognized by the
+//fuzzing:cc_engine_instrumentation and //fuzzing:cc_engine_sanitizer
+configuration flag should be defined here.
+"""
+
+def _check_is_string_list(name, value):
+    string_type = type("string")
+    list_type = type(["list"])
+
+    if type(value) != list_type:
+        fail("%s should be a list, but was %s" % (name, type(value)))
+    if any([type(element) != string_type for element in value]):
+        fail("%s should be a list of strings" % name)
+
+def instrumentation_opts(copts = [], linkopts = []):
+    """Creates a set of instrumentation options.
```

Comment:
nit: is there a word other than "set" that can be used here? When I read set I think of `{one, of, these, sets}`, but this isn't really returning a set.

If there's a way to introduce new types in bazel (i.e., the equivalent of `struct InstrumentationOptions { ... }` in C) then it would be easier to write the doc string here as just as `Creates a new InstrumentationOptions`

Maybe just struct? Or """Creates a new instrumentation options value in the format expected by...""" (this might be too wordy)

Reply:
I went with "struct". Unfortunately, Starlark doesn't support custom types, so "structs" are the best choice for structured data.